### PR TITLE
Implement 'replaces' field in newOrder and draft-ietf-acme-ari-03 CertID changes

### PR DESCRIPTION
**Pull request by beautifulentropy**
- Implement the updated CertID formation, as per draft-ietf-acme-ari-03.
- Implement the "replaces" field in newOrder for indicating an order is a replacement at renewal time, as per draft-ietf-acme-ari-03.
- Discontinue and remove the POST method implementation for indicating replacement success, deprecated in draft-ietf-acme-ari-03.

https://datatracker.ietf.org/doc/draft-ietf-acme-ari/

---

**PLEASE NOTE:** this MUST NOT be merged until https://github.com/letsencrypt/boulder/pull/7298 is deployed to Production. I'll keep this issue up-to-date.

### CODE REVIEW

**Comment by beautifulentropy** [comment]
This is expected and has occurred in the two previous ARI PRs that touch renewForDomains.

```
Error: cmd/cmd_renew.go:128:1: cyclomatic complexity 14 of func `renewForDomains` is high (> 12) (gocyclo)
func renewForDomains(ctx *cli.Context, client *lego.Client, certsStorage *CertificatesStorage, bundle bool, meta map[string]string) error {
```

**Comment by ldez** [comment]
The current failure of the CI is related to a Pebble bug, I fixed the problem inside #2119.
So I will just rebase your PR to pass the CI.

**Comment by beautifulentropy** [comment]
@ldez the changes necessary to support this contribution have been deployed at Let's Encrypt. Let me know if you need anything else.

**Review by ldez — APPROVED** [review]
Thank you!

**Comment by ldez** [comment]
I missed the conflict :cry:

File: certificate/renewal.go [hunk 1]
```diff
@@ -84,39 +84,32 @@ func (c *Certifier) GetRenewalInfo(req RenewalInfoRequest) (*RenewalInfoResponse
...
+		return "", errors.New("invalid DER encoding of serial number")
 	}
 
-	return fmt.Sprintf("%s.%s",
-		strings.TrimRight(base64.URLEncoding.EncodeToString(leaf.AuthorityKeyId), "="),
-		strings.TrimRight(base64.URLEncoding.EncodeToString(leaf.SerialNumber.Bytes()), "="),
-	), nil
+	// Extract only the integer bytes from the DER encoded Serial Number
+	// Skipping the first 2 bytes (tag and length).
+	serial := base64.RawURLEncoding.EncodeToString(der[2:])
```
Comment (mholt) [inline certificate/renewal.go:107]: @beautifulentropy Just curious, why the ASN.1 marshaling and byte stripping? In my testing, `leaf.SerialNumber.Bytes()` yields the same output as the more complex ASN.1 algorithm.

**Pull request merged**